### DISCUSSION TITLE
Adding Developer Certificate of Origin (DCO) GitHub action

### DISCUSSION THREAD

**Discussion by heyanlong** [apache/skywalking#11647] [Ideas]
SkyWalking is an open source product released under the Apache 2.0 license (see either [the Apache site](https://www.apache.org/licenses/LICENSE-2.0) or the [LICENSE file](https://github.com/apache/skywalking/blob/master/LICENSE)). The Apache 2.0 license allows you to freely use, modify, distribute, and sell your own products that include Apache 2.0 licensed software.

We respect intellectual property rights of others, and we want to make sure all incoming contributions are correctly attributed and licensed. A Developer Certificate of Origin (DCO) is a lightweight mechanism to do that.

The DCO is a declaration attached to every contribution made by every developer. In the commit message of the contribution, the developer simply adds a `Signed-off-by` statement and thereby agrees to the DCO.

It can assist contributors in effectively managing code copyright even if they haven't signed a [ICLA or CCLA](https://www.apache.org/licenses/contributor-agreements.html).

About the DCO bot https://probot.github.io/apps/dco/

**Comment by wu-sheng** [comment]
DCO thing is common in CNCF/Linux Foundation. But I don't know the requirements and opinions of the ASF.
Is there any reason you are asking for this?

**Comment by heyanlong** [comment]
It appears to me that the majority of contribution have not signed the ICLA. This can serve as an addition. All you have to do to commit is add the `-s` option.

**Reply by wu-sheng** [reply]
I mean, this is a very high level requirements.
You are a committer, so, you had signed icla.

**Reply by heyanlong** [reply]
But for other non-committers, it seems that they are not signed

**Reply by wu-sheng** [reply]
Yes, they are not, because the ASF doesn't require that.
If we require DCO, this is a hard requirement, we need the confirmation from the foundation.

**Reply by heyanlong** [reply]
OK, I understand what you mean

**Comment by kezhenxu94** [comment]
I have no opinion to or not to add this GitHub Actions, but if it will be added, I'd suggest we make it optional, DCO is only one lightweight way to verify the author of the commits, but for example, I prefer to sign the entire commit with GPG key and all my previous commits have been signed, and I don't like to add those (stupid) lines "Signed-off-by" in my commit messages
<img width="949" alt="image" src="https://github.com/apache/skywalking/assets/15965696/e59da8cf-6f67-4bec-9d28-d71a1c6f0334">
